### ISSUE TITLE
Dart 2 Incompatible 

### ISSUE THREAD

**Issue by mogol** [denniskaselow/dartemis#52]
Hello, 
[the latest available version ](https://pub.dev/packages/dartemis/versions) in pub.dev is dart 2 incompatible. However all tests are green for the dart 2.7  in the latest master. Could it be published?

Thanks

**Comment by denniskaselow** [comment]
I'm surprised someone else than me is actually using this after all this time :).

A while ago I did some major changes to the API in the entityToInt-branch which I am using in my current projects. Are you already using dartemis for something? Would you rather have the master branch published where there is still an `Entity` class or would you be fine with the version where an entity is just a simple `int`?

**Comment by mogol** [comment]
I have just started experimenting with ECS for my future project (game server, aot/jit dart) and was very happy to find this project 👍 thanks.

So it's fine by me to use int :) wouldn't break anything. However I am in the early stage to provide any feedback on the change. 

Out of curiosity, why are these changes? performance/memory?

**Comment by mogol** [comment]
@denniskaselow It would be great if the corresponding version of the builder is published as well

**Comment by denniskaselow** [comment]
Yes, it was for performance reasons. With the `Entity` approach I had to update several lists/bags of them whenever a new `Entity` was added to or removed from the game, which meant that this was only done after one frame was processed. But I often had a use case, where I wanted to create an `Entity` in one system and use it in another one immediately afterwards, so initially I added a way to trigger the process of updating those lists, but this was error prone (if you forgot to trigger it) and it was bad for he performance the more often I needed to do that.

Now the "Entity" is just the index of a bit in a `BitSet` and is updated by doing some `OR`-ind and `AND`-ing.

I've also been working on a NNBD version of dartemis, so whenever that feature hits the stable version of Dart, I can publish an updated version soon after.

I'll publish dartemis and the builder in the next 12 hours. Just need to check if there is anything more I need to cleanup before I do.
